app/models/host/base.rb

```diff
@@ -321,6 +323,26 @@ def import_missing_ids
     # Provide _id aliases for consistency with the _name methods
     alias_attribute :hardware_model_id, :model_id
 
+    def locked?
```

Comment:
not sure I understand what you're asking. this is basically how this is implemented, the locked_until method returns a timestamp which represents the time the host gets unlocked (or -1 if the host is locked without a time limit)